synapse/events/validator.py

```diff
@@ -87,6 +90,55 @@ def validate_new(self, event: EventBase, config: HomeServerConfig):
                     400, "Can't create an ACL event that denies the local server"
                 )
 
+        if event.type == EventTypes.PowerLevels:
+            powerLevelsSchema = {
+                "type": "object",
+                "properties": {
+                    "ban": {"$ref": "#/definitions/int"},
+                    "events": {"$ref": "#/definitions/objectOfInts"},
+                    "events_default": {"$ref": "#/definitions/int"},
+                    "invite": {"$ref": "#/definitions/int"},
+                    "kick": {"$ref": "#/definitions/int"},
+                    "notifications": {"$ref": "#/definitions/objectOfInts"},
+                    "redact": {"$ref": "#/definitions/int"},
+                    "state_default": {"$ref": "#/definitions/int"},
+                    "users": {"$ref": "#/definitions/objectOfInts"},
+                    "users_default": {"$ref": "#/definitions/int"},
+                },
+                "definitions": {
+                    "int": {
+                        "type": "integer",
+                        "minimum": -9007199254740991,
+                        "maximum": 9007199254740991,
```

Comment:
could I ask you to factor these out to constants somewhere (eg, `CANONICALJSON_{MIN,MAX}_INT`)? There is another reference in `synapse.events.utils` which could do with de-duplicating.